### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by alvra**
Changes:
  * Session Renewal
  * Handle change of remote type
  * Allow interaction in unknown network state

### CODE REVIEW

**Comment by gijzelaerr** [comment]
fails quite quickly on my side ```parallels@bloempot:~/Work/surfnet/python-eduvpn-client$ make eduvpn-gui 
./venv/bin/eduvpn-gui
(eduvpn-gui:467799): Gtk-WARNING **: 15:24:02.837: Overriding tab label for notebook
(eduvpn-gui:467799): Gtk-WARNING **: 15:24:02.838: Overriding tab label for notebook
2021-03-17 15:24:02,842 - MainThread - INFO - eduvpn.ui.ui - ui.py:155 - starting ui
2021-03-17 15:24:02,847 - init-server-db - INFO - eduvpn.remote - remote.py:23 - Requesting https://disco.eduvpn.org/v2/server_list.json
/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/nm.py:34: Warning: g_main_context_push_thread_default: assertion 'acquired_context' failed
  return NM.Client.new(None)
/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/nm.py:34: Warning: g_main_context_pop_thread_default: assertion 'g_queue_peek_head (stack) == context' failed
  return NM.Client.new(None)
2021-03-17 15:24:02,913 - MainThread - INFO - eduvpn.app - app.py:111 - network transitioning: <InitialNetworkState> -> found_active_connection
2021-03-17 15:24:02,913 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <InitialInterfaceState> -> found_active_connection
Traceback (most recent call last):
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/app.py", line 115, in network_transition
    self.network_state_machine.transition(
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/state_machine.py", line 139, in transition
    new_state = transition_func(*args, **kwargs)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/network.py", line 81, in found_active_connection
    app.interface_transition('found_active_connection', server, validity)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/app.py", line 134, in interface_transition
    self.interface_state_machine.transition(
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/state_machine.py", line 143, in transition
    self._call_edge_callbacks(ENTER, old_state, new_state)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/state_machine.py", line 192, in _call_edge_callbacks
    callback(old_state, new_state)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/ui/ui.py", line 456, in enter_ConnectionStatus
    self.update_connection_status()
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/ui/ui.py", line 235, in update_connection_status
    self.update_connection_validity()
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/ui/ui.py", line 220, in update_connection_validity
    expiry_text = get_expiry_text(self.app.interface_state.validity.end)
AttributeError: 'NoneType' object has no attribute 'end'
2021-03-17 15:24:03,965 - init-server-db - INFO - eduvpn.remote - remote.py:37 - Retrieving signature https://disco.eduvpn.org/v2/server_list.json.minisig
2021-03-17 15:24:04,633 - init-server-db - INFO - eduvpn.remote - remote.py:44 - verifying signature of https://disco.eduvpn.org/v2/server_list.json
2021-03-17 15:24:04,634 - init-server-db - INFO - eduvpn.remote - remote.py:23 - Requesting https://disco.eduvpn.org/v2/organization_list.json
2021-03-17 15:24:05,954 - init-server-db - INFO - eduvpn.remote - remote.py:37 - Retrieving signature https://disco.eduvpn.org/v2/organization_list.json.minisig
2021-03-17 15:24:06,627 - init-server-db - INFO - eduvpn.remote - remote.py:44 - verifying signature of https://disco.eduvpn.org/v2/organization_list.json
2021-03-17 15:24:06,632 - MainThread - INFO - eduvpn.app - app.py:130 - interface transitioning: <ConnectionStatus server=<InstituteAccessServer 'Demo'>, validity=None> -> server_db_finished_loading
Traceback (most recent call last):
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/app.py", line 134, in interface_transition
    self.interface_state_machine.transition(
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/state_machine.py", line 140, in transition
    self._call_edge_callbacks(EXIT, old_state, new_state)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/state_machine.py", line 192, in _call_edge_callbacks
    callback(old_state, new_state)
  File "/home/parallels/Work/surfnet/python-eduvpn-client/eduvpn/ui/ui.py", line 486, in exit_ConnectionStatus
    self._cancel_validity_updates()
AttributeError: 'EduVpnGui' object has no attribute '_cancel_validity_updates'```

File: eduvpn/server.py [hunk 1]
```diff
@@ -196,6 +198,13 @@ def support_contact(self) -> List[str]:
 
     @property
     def oauth_login_url(self):
+        print('=' * 20)
+        from pprint import pprint
```
Comment (gijzelaerr) [inline eduvpn/server.py:202]: i think this for debugging? we should not print to the console, unless its for the CLI.

File: eduvpn/ui/ui.py [hunk 2]
```diff
@@ -447,36 +459,37 @@ def enter_ConnectionStatus(self, old_state, new_state):
...
+            'update-validity',
         )
 
     @transition_edge_callback(EXIT, interface_state.ConnectionStatus)
     def exit_ConnectionStatus(self, old_state, new_state):
         self.show_component('connectionPage', False)
 
-        self._cancel_expiry_updates()
-        del self._cancel_expiry_updates
+        if hasattr(self, '_cancel_validity_updates'):
```
Comment (gijzelaerr) [inline eduvpn/ui/ui.py:490]: this looks a bit like a hack, isn't there another way to accomplish this? With a boolean flag? I just find a bit odd the change the signature of an object runtime.

File: eduvpn/server.py [hunk 3]
```diff
@@ -312,5 +328,7 @@ def search(self, query: str) -> Iterable[PredefinedServer]:
             yield from self.all()
 
     def get_server_info(self, server) -> ServerInfo:
+        # try:
         info = remote.get_info(server.oauth_login_url)
+        # except requests.exceptions.ConnectionError:
```
Comment (gijzelaerr) [inline eduvpn/server.py:333]: did you forget to remove this debug code?

**Review by gijzelaerr — CHANGES_REQUESTED** [review]

**Comment by alvra** [comment]
Removed accidental leftover debug lines.

File: eduvpn/ui/ui.py (hunk 2 above)
Comment (alvra) [reply eduvpn/ui/ui.py:490]: Yes, this bit is not pretty. I will create a proper mechanism for this later.

**Pull request merged**